### PR TITLE
Add progression mechanism and progression indicator into views

### PR DESCRIPTION
**Pull request by gregorylegarec**
Will not work without #307. I tried with this PR to push a rebased version of my branch to isolate only commits related to the progression feature.

This PR adds the method `getProgression(step)` to Onboarding class.

It also implements the logic to get it displayed into views.

## Important changes

This PR introduce a StepView "abstract" class. All step views now inherit from this class.

The markup used in views for progression indicator is the same than the proposed one in @m4dz's PR #305. So all step view templates are changing a little bit but it is absolutely not their final version.

Also added :
* Tests
* Documentation

### CODE REVIEW

**Comment by m4dz** [comment]
@gregorylegarec can you please rebase it on development now #307 is merged in? I'll let the review be more obvous 😄

**Comment by gregorylegarec** [comment]
@m4dz rebased.

File: client/app/views/steps/subviews/progression.coffee [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{LayoutView} = require 'backbone.marionette'
+
+module.exports = class ProgressionView extends LayoutView
```
Comment (m4dz) [inline client/app/views/steps/subviews/progression.coffee]: Why using a layoutview instead of a simple Marionette.View here?

File: client/app/views/templates/view_steps_agreement.jade [hunk 2]
```diff
@@ -2,4 +2,7 @@ extends view_base
 
 block region
     h1=t('Conditions générales')
```
Comment (m4dz) [inline client/app/views/templates/view_steps_agreement.jade]: We must use key strings for localization. Do we doit here, or do your prefer to open another PR to declare those string keys?

File: client/app/lib/onboarding.coffee [hunk 3]
```diff
@@ -119,5 +119,17 @@ module.exports = class Onboarding
...
+
+    # Returns progression associated to the given step object
+    # @param step Step which we want to know the related progression
+    # returns the current index of the step, from 1 to length. 0 if the step
+    # does not exist in the onboarding.
+    getProgression: (step) ->
+        return \
+            current: @steps.indexOf(step)+1,
+            total: @steps.length,
+            labels: @steps.map (step) -> step.name
```
Comment (m4dz) [inline client/app/lib/onboarding.coffee]: I can't figure out why `getProgression` is returning a js object instead of a `ProgressionModel` object. What do I miss here?

File: client/app/views/steps/subviews/progression.coffee (hunk 1 above)
Comment (gregorylegarec) [inline client/app/views/steps/subviews/progression.coffee]: The very first view was a copy-paste from another one, and LayoutView stayed.

I am changing it right now.

Comment (gregorylegarec) [inline client/app/views/steps/subviews/progression.coffee]: Ok, actually it's not working with a View, but it does with an ItemView. What do you think is better ?

File: client/app/views/templates/view_steps_agreement.jade (hunk 2 above)
Comment (gregorylegarec) [inline client/app/views/templates/view_steps_agreement.jade]: Maybe we could do this during markup integration ? Until now those labels are purely indicative.

File: client/app/lib/onboarding.coffee (hunk 3 above)
Comment (gregorylegarec) [inline client/app/lib/onboarding.coffee]: Same as Step Object, I try to keep the return values from any Onboarding method as agnostic as possible. ProgressionModel is a pure wrapper for using this result into a Backbone/Marionette View.

File: client/app/views/steps/subviews/progression.coffee (hunk 1 above)
Comment (m4dz) [inline client/app/views/steps/subviews/progression.coffee]: `ItemView` is the regular way offered by Marionette for _simple_ views. So if we want to to it is a proper way, I recommend this one

File: client/app/views/templates/view_steps_agreement.jade (hunk 2 above)
Comment (m4dz) [inline client/app/views/templates/view_steps_agreement.jade]: Perfect!

File: client/app/lib/onboarding.coffee (hunk 3 above)
Comment (m4dz) [inline client/app/lib/onboarding.coffee]: Oh, true. I understand, thanks 😃

**Comment by m4dz** [comment]
Thanks! 🍿
